src/db/models/device.rs

```diff
@@ -212,4 +212,16 @@ impl Device {
                 .from_db()
         }}
     }
+
+    pub async fn check_user_has_push_device(user_uuid: &str, conn: &mut DbConn) -> bool {
+        db_run! { conn: {
+            diesel::select(diesel::dsl::exists(
+                devices::table
+                    .filter(devices::user_uuid.eq(user_uuid))
+                    .filter(devices::push_token.is_not_null())
+            ))
+            .get_result(conn)
+            .expect("Error loading push devices")
```

Comment:
This is probably better, since it will return true if there is a count of 1 or higher, and false otherwise.
No need for errors here.

```suggestion
            .count()
            .first::<i64>(conn)
            .ok()
            .unwrap_or(0) != 0
```